include/mbedtls/ssl.h

```diff
@@ -382,6 +388,28 @@ union mbedtls_ssl_premaster_secret
 
 #define MBEDTLS_PREMASTER_SIZE     sizeof( union mbedtls_ssl_premaster_secret )
 
+static inline int mbedtls_ssl_chk_buf( uint8_t *cur, const uint8_t *end, size_t need )
+{
+    return( end < cur || (size_t)( end - cur ) < (size_t) need );
+}
+
+#define MBEDTLS_SSL_CHK_BUF_PTR( cur, end, need )			\
+    do {								\
+        if( mbedtls_ssl_chk_buf( ( cur ), ( end ), ( need ) ) != 0 )	\
+        {								\
+            return( MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL );			\
+        }								\
+    } while( 0 )
```

Comment:
@andresag01 That changes the debugging output in preexisting code, which I want to avoid. Note that sometimes test suites depend on the debugging output.